python/S3_BUCKET_NAMING_CONVENTION/S3_BUCKET_NAMING_CONVENTION_test.py

```diff
@@ -0,0 +1,176 @@
+except ImportError:
+    from mock import MagicMock
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::S3::Bucket'
+
+#############
+# Main Code #
+#############
+
+CONFIG_CLIENT_MOCK = MagicMock()
+STS_CLIENT_MOCK = MagicMock()
+S3_CLIENT_MOCK = MagicMock()
+
+class Boto3Mock():
+    @staticmethod
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        if client_name == 's3':
+            return S3_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('S3_BUCKET_NAMING_CONVENTION')
+
+class ComplianceTest(unittest.TestCase):
+
+    invoking_event_bucket = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"apps-us-east-1","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
+    invoking_event_bucket_compliant = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"apps-test-us-east-1","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
+    s3_bucket_list = {"Buckets": [{"Name": "852640994763-awsmacietrail-dataevent", "CreationDate": "2018-07-12T05:26:16.000Z"}]}
+
+    def setUp(self):
+        pass
+
+    def test_scenario_1a_not_given(self):
+        rule_param = "{\"regexPattern\":\"\"}"
+        invoking_event = self.invoking_event_bucket
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=rule_param)
+        response = RULE.lambda_handler(lambda_event, {})
+        assert_customer_error_response(self, response, "InvalidParameterValueException")
+
+    def test_scenario_1b_invalid(self):
+        rule_param = "{\"regexPattern\":\"[bad\"}"
+        invoking_event = self.invoking_event_bucket
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=rule_param)
+        response = RULE.lambda_handler(lambda_event, {})
+        assert_customer_error_response(self, response, "InvalidParameterValueException")
+
+    def test_scenario_2_non_compliant(self):
+        rule_param = "{\"regexPattern\":\".*test.*\"}"
+        S3_CLIENT_MOCK.list_buckets = MagicMock(return_value=self.s3_bucket_list)
```

Comment:
Mock of `list_buckets` not required as you have changed the rule code to not include this API call.